core/src/main/java/com/segment/analytics/kotlin/core/HTTPClient.kt

```diff
@@ -34,7 +34,7 @@ class HTTPClient {
 
     private fun authorizationHeader(writeKey: String): String {
         val auth = "$writeKey:"
-        return "Basic " + Base64.getEncoder().encodeToString(auth.toByteArray())
+        return "Basic ${encodeToBase64(auth)}"
```

Comment:
I feel like it may be a good idea to only re-encode when the writeKey changes instead of every time we ask for the authorizationHeader. Mainly because our `encodeToBase64` method is O(n^2). Thoughts?

Reply:
ooh good catch, lemme see if I can add that optimization in our codebase

Reply:
correct me if I'm wrong. I think our `encodeToBase64` method is still O(n). the inner for loops are capped by 3, so worst case O(3n). but still changing it to only re-encode when writeKey changes is a good idea! XD

Reply:
yea i havent done the full analysis, but wenxi might be right, its of the order O(n) rather than O(n^2). But anyways caching the header is a good idea since it does not actually change over time.

Reply:
I see it as O(n^2) because We have a while loop and inside the while loop 3n (3 for loops).

Reply:
ah. gotcha. yeah. that also makes sense. I was thinking the while loop takes m, the 3 inner loops each takes n (where n <= 3), thus the complexity is O(m*3n) <= O(m*3*3) = O(9m) = O(m). but it also make sense to treat the inner loop n as a variable, and indeed it is. so O(m*3n) is more accurate. 😄